javatests/arcs/core/data/proto/HandleProtoDecoderTest.kt

```diff
@@ -113,30 +115,127 @@ class HandleProtoDecoderTest {
         )
         with(handleProto.decode(handles)) {
             assertThat(name).isEqualTo("thing")
+            assertThat(id).isEqualTo("")
             assertThat(fate).isEqualTo(Handle.Fate.JOIN)
             assertThat(storageKey).isEqualTo("ramdisk://b")
             assertThat(associatedHandles).isEqualTo(listOf(handles["handle1"], handles["handle_join"]))
             assertThat(type).isEqualTo(entityType)
+            assertThat(tags).isEmpty()
             assertThat(capabilities).isEqualTo(Capabilities.PersistentQueryable)
         }
     }
 
+    @Test
+    fun decodesHandleProtoWithTypeAndTags() {
+        val entityTypeProto =
+            """
+              entity {
+                schema {
+                  names: "Thing"
+                  fields {
+                    key: "name"
+                    value: { primitive: TEXT }
+                  }
+                }
+              }
+            """.trimIndent()
+        val storageKey = "ramdisk://b"
+        val entityType = parseTypeProtoText(entityTypeProto).decode()
+        val handleText = buildHandleProtoText(
```

Comment:
Given the number of arguments to this function, it would be more readable to use name parameters:
```
buildHandleProtoText(
    name = "thing",
    fate = "JOIN",
    ...
)
```
Similarly for other use cases that you added?